obspy/signal/cross_correlation.py

```diff
@@ -26,14 +26,168 @@
 
 import numpy as np
 import scipy
+from scipy.signal import correlate, fftconvolve
 
 from obspy import Stream, Trace
 from obspy.core.util.misc import MatplotlibBackend
 from obspy.signal.headers import clibsignal
 from obspy.signal.invsim import cosine_taper
 
 
-def xcorr(tr1, tr2, shift_len, full_xcorr=False):
+def _pad_zeros(a, num):
+    """Pad num zeros at both sides of array a"""
+    hstack = [np.zeros(num), a, np.zeros(num)]
+    return np.hstack(hstack)
+
+
+def _xcorr_padzeros(a, b, num, domain='freq'):
+    """
+    Cross-correlation using SciPy with mode='valid' and precedent zero padding
+    """
+    if num is None:
+        num = (len(a) + len(b) - 1) // 2
+    dif = len(a) - len(b) - 2 * num
+    if dif > 0:
+        b = _pad_zeros(b, dif // 2)
+    else:
+        a = _pad_zeros(a, -dif // 2)
+    if domain == 'freq':
+        c = fftconvolve(a, b[::-1], 'valid')
+    else:
+        c = correlate(a, b, 'valid')
+    return c
+
+
+def _xcorr_slice(a, b, num, domain='freq'):
+    """
+    Cross-correlation using SciPy with mode='full' and subsequent slicing
+    """
+    if domain == 'freq':
+        c = fftconvolve(a, b[::-1], 'full')
+    else:
+        c = correlate(a, b, 'full')
```

Comment:
Please also make this typo safe, see other comment.. (if/elif/else raise ValueError)